### PR TITLE
feat(ui): Allow project selector on project detail page

### PR DESCRIPTION
**Pull request by matejminar**
This PR enables switching projects in the global header on the project detail page.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx [hunk 1]
```diff
@@ -35,11 +40,13 @@ type RouteParams = {
 
 type Props = RouteComponentProps<RouteParams, {}> & {
   organization: Organization;
+  // HoC vvv
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx:43]: Please what does mean `vvv`?

File: src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx [hunk 2]
```diff
@@ -122,17 +198,25 @@ class ProjectDetail extends AsyncView<Props, State> {
...
-                ))}
-                <ProjectIssues organization={organization} location={location} />
+                <ProjectScoreCards
+                  organization={organization}
+                  isProjectStabilized={isProjectStabilized}
+                  selection={selection}
+                />
+                {isProjectStabilized && (
+                  <React.Fragment>
+                    {[0, 1].map(id => (
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx:208]: Please why are we using an array of numbers here? why length 2?

File: src/sentry/static/sentry/app/views/projectDetail/projectLatestReleases.tsx [hunk 3]
```diff
@@ -27,6 +27,7 @@ type Props = AsyncComponent['props'] & {
   projectSlug: string;
   location: Location;
   projectId?: string;
+  isProjectStabilized: boolean;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectDetail/projectLatestReleases.tsx:30]:
```suggestion
  isProjectStabilized: boolean;
  projectId?: string;
```

File: src/sentry/static/sentry/app/views/projectDetail/projectLatestReleases.tsx [hunk 4]
```diff
@@ -36,19 +37,38 @@ type State = {
...
     }
 
     return false;
   }
 
+  componentDidUpdate(prevProps: Props) {
+    const {location, isProjectStabilized} = this.props;
+
+    if (
+      prevProps.location.query.environment !== location.query.environment ||
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectDetail/projectLatestReleases.tsx:58]: we repeat this code in a couple of files. Should we create a reusable function?

File: tests/js/spec/views/projectDetail/projectLatestReleases.spec.jsx [hunk 5]
```diff
@@ -95,6 +97,7 @@ describe('ProjectDetail > ProjectLatestReleases', function () {
           projectSlug={project.slug}
           location={router.location}
           projectId={project.slug}
+          isProjectStabilized
```
Comment (priscilawebdev) [inline tests/js/spec/views/projectDetail/projectLatestReleases.spec.jsx:100]: should we also test when `isProjectStabilized` equals false?

File: src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx [hunk 6]
```diff
@@ -48,26 +55,95 @@ class ProjectDetail extends AsyncView<Props, State> {
...
+
+    return projects.find(p => p.slug === params.projectId);
+  }
 
-    if (this.state?.project) {
-      return [];
+  handleProjectChange = (selectedProjects: number[]) => {
+    const {projects, router, location, organization} = this.props;
+
+    const newlySelectedProject = projects.find(p => p.id === String(selectedProjects[0]));
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx:75]: will the project details page be updated according to the first selected project?

File: src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx (hunk 1 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx]: It's to indicate that props downwards come from HoC but I guess we don't need it 👍

File: src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx (hunk 2 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx:207]: Right now we show two charts on the project detail page - we are experimenting with showing more/fewer charts there and this allows us to do so.

File: src/sentry/static/sentry/app/views/projectDetail/projectLatestReleases.tsx (hunk 4 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/projectDetail/projectLatestReleases.tsx]: done 👍

File: tests/js/spec/views/projectDetail/projectLatestReleases.spec.jsx (hunk 5 above)
Comment (matejminar) [reply tests/js/spec/views/projectDetail/projectLatestReleases.spec.jsx:100]: yep, I meant to add them in a follow-up but pushed them here now 👍

File: src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx (hunk 6 above)
Comment (matejminar) [reply src/sentry/static/sentry/app/views/projectDetail/projectDetail.tsx:74]: you can only select one on this page

File: src/sentry/static/sentry/app/views/projectDetail/projectLatestAlerts.tsx [hunk 7]
```diff
@@ -37,19 +39,36 @@ type State = {
...
 
     return false;
   }
 
+  componentDidUpdate(prevProps: Props) {
+    const {location, isProjectStabilized} = this.props;
+
+    if (
+      didProjectOrEnvironmentChange(prevProps.location, location) ||
+      prevProps.isProjectStabilized !== isProjectStabilized
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/projectDetail/projectLatestAlerts.tsx:60]: should we include `isProjectStabilized` in the function too?

Comment (matejminar) [reply src/sentry/static/sentry/app/views/projectDetail/projectLatestAlerts.tsx:60]: I'd rather not - let's keep it more explicit.

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**